tls.go

```diff
@@ -174,3 +176,43 @@ func loadX509KeyPair(certFile, keyFile string) (tls.Certificate, error) {
 	}
 	return tls.X509KeyPair(certPEMBlock, keyPEMBlock)
 }
+
+func reportTLSCertsExpiration(promConfig PromConfig, tlsConfig *tls.Config) error {
+	for _, cert := range tlsConfig.Certificates {
+		x509cert, err := leaf(&cert)
+		if err != nil {
+			return err
+		}
+
+		if err := reportCertExpiration(promConfig, x509cert); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func reportCertExpiration(promConfig PromConfig, cert *x509.Certificate) error {
+	d := time.Until(cert.NotAfter) / (24 * time.Hour)
+	return registerCertExpirationMetric(
+		promConfig, cert.Subject.CommonName, strings.Join(cert.DNSNames, ","), strings.Join(cert.Subject.Organization, ","), int(d),
+	)
+}
+
+func leaf(cert *tls.Certificate) (*x509.Certificate, error) {
+	if cert.Leaf != nil {
+		return cert.Leaf, nil
+	}
+	return x509.ParseCertificate(cert.Certificate[0])
+}
+
+func registerCertExpirationMetric(promConfig PromConfig, cn, dnsNames, organization string, remainingDays int) error {
+	return promConfig.PromRegistry.Register(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Namespace:   promConfig.PromNamespace,
+		Name:        "days_until_cert_expiring",
+		Help:        "Number of days until the certificate expires",
+		ConstLabels: prometheus.Labels{"cn": cn, "dns_names": dnsNames, "organization": organization},
+	}, func() float64 {
+		return float64(remainingDays)
```

Comment:
Could we calculate remainingDays here? That way the value will be updated on each day.